node-ram-cache.cpp

```diff
@@ -107,15 +109,13 @@ ramNode *node_ram_cache::next_chunk() {
 }
 
 
-int node_ram_cache::set_sparse(osmid_t id, const ramNode &coord) {
+void node_ram_cache::set_sparse(osmid_t id, const ramNode &coord) {
     // Sparse cache depends on ordered nodes, reject out-of-order ids.
     // Also check that there is still space.
     if ((maxSparseId && id < maxSparseId)
         || (sizeSparseTuples > maxSparseTuples)
         || ( cacheUsed > cacheSize)) {
-        if ((allocStrategy & ALLOC_LOSSY) > 0)
-            return 1;
-        else {
+        if (!((allocStrategy & ALLOC_LOSSY) > 0)) {
```

Comment:
I think I broke the logic here. Before, if it met one of the out of memory or out of order conditions, it would always exit from this function, either by returning or by util::exit_nicely(). Now, it can go to the later on logic in the right condition